static_frame/core/index_hierarchy.py

```diff
@@ -2784,19 +2780,21 @@ def level_add(self: IH,
         if self._recache:
             self._update_array_cache()
 
-        index_cls = self._INDEX_CONSTRUCTOR if index_constructor is None else index_constructor._MUTABLE_CONSTRUCTOR # type: ignore
+        if index_constructor is None:
+            index_cls = self._INDEX_CONSTRUCTOR
+        else:
+            index_cls = index_constructor._IMMUTABLE_CONSTRUCTOR if self.STATIC else index_constructor._MUTABLE_CONSTRUCTOR # type: ignore
 
         if self.STATIC:
             indices = [index_cls((level,)), *self._indices]
         else:
             indices = [index_cls((level,)), *(idx.copy() for idx in self._indices)]
 
-        indexers = np.array(
-                [
-                    np.zeros(self.__len__(), dtype=DTYPE_INT_DEFAULT),
-                    *self._indexers
-                ],
-                dtype=DTYPE_INT_DEFAULT,
+        indexers = np.vstack(
+            (
+                np.zeros(self.__len__(), dtype=DTYPE_INT_DEFAULT),
+                self._indexers,
+            )
```

Comment:
This might be more efficient:
```suggestion
        indexers = np.zeros(len(self._indexers) + 1, self.__len__(), dtype=DTYPE_INT_DEFAULT)
        indexers[1:] = self._indexers 
```